### ISSUE TITLE
Factory classes for CombineHarvester

### ISSUE THREAD

**Issue by ajgilbert** [cms-analysis/HiggsAnalysis-HiggsToTauTau#609]
AddBinByBin, MergeBinByBin, AddSyst and CreateRooMorphingPdf should become factory classes so that they are easily extensible by others. This means we need to implement direct push_back of new Objects. Once done the existing methods should be deprecated.

**Comment by ajgilbert** [comment]
Would be nice to have something like the 'split' mode of the official nuisance editing tool. The implementation would be something like:

~~~cpp
CopySysts(cmb.cp().syst_name({"lumi_7TeV"}), cmb, [](Systematic *s) {
  s->set_name("lumi_2011"); s->set_value_u(1.006);
});
CopySysts(cmb.cp().syst_name({"lumi_7TeV"}), cmb, [](Systematic *s) {
  s->set_name("CMS_lumi_2011"); s->set_value_u(1.0212);
});
cmb.syst_name({"lumi_7TeV"}, false);
~~~

Which could always be wrapped in another function:
~~~cpp
SplitSyst(cmb, "lumi_7TeV", "lumi_2011", "CMS_lumi_2011", 1.006, 1.0212);
~~~

**Comment by ajgilbert** [comment]
resolved by #627
